__tests__/ExpensiMark-HTML-test.js

```diff
@@ -2220,7 +2220,7 @@ describe('room mentions', () => {
 
     test('room mention inside code block should not be rendered', () => {
         const testString = '```\n#room\n```';
-        const resultString = '<pre>#room<br /></pre>';
+        const resultString = '<pre><br />#room<br /></pre>';
```

Comment:
I believe this is unwanted. HTML BR collapsing works for `<br /></pre>` but not for `<pre><br />` https://codesandbox.io/s/silly-brattain-dz3rrw

Reply:
Interesting. The App PreRenderer works fine though, but I think we may want to avoid this. Maybe we can put the new line info on the <pre> tag like:
```
<pre data-newline="\r\n">
```

and live markdown can get the new line from `data-newline`.

Reply:
That would work but it won't work if we have mixed newlines. Technically the live markdown works correctly and the console error is a false positive. Maybe we can ignore the error if the difference between the original text and the reconstructed one is only the new lines? Though ultimately we want the reconstructed message to be the same

Reply:
> That would work but it won't work if we have mixed newlines

Hmm, you're right.

> Maybe we can ignore the error if the difference between the original text and the reconstructed one is only the new lines?

I think we can replace all the new lines to \n only when doing the comparison. This way, we don't alter the user input at all, but we need @BartoszGrajdek thoughts on this one.

Reply:
What if we added the line break just in `rawInputReplacement`? The returned value of it is coming only into `react-native-live-markdown` input, so it won't affect how we treat code blocks in the chat itself, just the input. 

We could just treat the linebreak before code block content as a separate group and use it only in `rawInputReplacement`.

Reply:
Something like this:
```
replacement: (_extras, _match, _g1, _g2, textWithinFences) => {
    const group = textWithinFences.replace(/(?:(?![\n\r])\s)/g, '&#32;');
    return `<pre>${group}</pre>`;
},
rawInputReplacement: (_extras, _match, _g1, g2, textWithinFences) => {
    const group = textWithinFences.replace(/(?:(?![\n\r])\s)/g, '&#32;').replace(/<emoji>|<\/emoji>/g, '');
    return `<pre>${g2}${group}</pre>`;
},

Reply:
Yeah, that works.

@s77rt do you agree with the above suggestion?

Reply:
Sounds good to me but can you please test this against `react-native-live-markdown` and verify that we don't get extra newlines

Reply:
Yeah, I don't see any extra newlines
<img width="925" alt="image" src="https://github.com/Expensify/expensify-common/assets/50919443/e025d7f2-5995-41c7-b2ed-f65b21233bd4">

Reply:
Let's go with that then 👍

Reply:
Updated!